src/psyclone/psyGen.py

```diff
@@ -5396,6 +5408,48 @@ def iterateitems(nodes):
                                 "declarations for fparser nodes {1}."
                                 "".format(str(arg_list), nodes))
 
+        # fparser2 does not always handle Statement Functions correctly, this
+        # loop checks for Stmt_Functions that should be an array statement
+        # and recovers them, otherwise it raises an error as currently
+        # Statement Functions are not supported in PSyIR.
+        for stmtfn in walk_ast(nodes, [Fortran2003.Stmt_Function_Stmt]):
+            (fn_name, arg_list, scalar_expr) = stmtfn.items
+            try:
+                symbol = parent.symbol_table.lookup(fn_name.string)
+                if symbol.is_array:
+                    # This is an array assignment wrongly categorized as a
+                    # statement_function by fparser2.
+                    array_name = fn_name
+                    if hasattr(arg_list, 'items'):
+                        array_subscript = arg_list.items
+                    else:
+                        array_subscript = [arg_list]
+                    assignment_rhs = scalar_expr
```

Comment:
pylint points out that `assignment_rhs` is not used.

Reply:
ups fixed